plugins/ui/src/js/src/elements/spectrum/TextField.tsx

```diff
@@ -3,31 +3,53 @@ import {
   TextField as DHCTextField,
   TextFieldProps as DHCTextFieldProps,
 } from '@deephaven/components';
-import { useDebouncedCallback } from '@deephaven/react-hooks';
+import { useDebouncedCallback, usePrevious } from '@deephaven/react-hooks';
 
 const VALUE_CHANGE_DEBOUNCE = 250;
 
-const EMPTY_FUNCTION = () => undefined;
+interface TextFieldProps extends DHCTextFieldProps {
+  onChange: (value: string) => Promise<void>;
```

Comment:
This still needs to be marked as optional, as you could omit it
```suggestion
  onChange?: (value: string) => Promise<void>;
```
Right now the following is throwing errors when you change the field, as there's no on_change handler:
```python
from deephaven import empty_table, ui
x = ui.text_field(value="Hello")
```

You don't _have_ to provide an `on_change` listener, you could have the text_field as part of a form to get input, e.g.

```python
@ui.component
def ui_form_submit():
    def handle_submit(data):
        print(f"Hello {data['name']}, you are {data['age']} years old")

    return ui.form(
        ui.text_field(default_value="Douglas", name="name"),
        ui.number_field(default_value=42, name="age"),
        ui.button("Submit", type="submit"),
        on_submit=handle_submit,
    )


my_form_submit = ui_form_submit()
```